src/script/lua_api/l_localplayer.cpp

```diff
@@ -430,14 +430,15 @@ int LuaLocalPlayer::l_hud_get_elements(lua_State *L)
 		return 0;
 
 	lua_newtable(L);
-	u32 max = player->getHudIdMax();
-	for (u32 id = 0; id < max; id++) {
-		HudElement *elem = player->getHud(id);
-		if (elem != nullptr) {
-			push_hud_element(L, elem);
-			lua_rawseti(L, -2, id);
+	player->hudApply([&](const std::vector<HudElement*>& hud) {
+		for (std::size_t id = 0; id < hud.size(); ++id) {
+			HudElement *elem = hud[id];
+			if (elem != nullptr) {
```

Comment:
Can the elements still be nullptrs if you're iterating over the vector of active elements?

Reply:
Yes, they are nullptrs when they get removed.
https://github.com/minetest/minetest/blob/d1a55e9ca4e66d8ba2d799569de7a6b48bc15f23/src/player.cpp#L142-L152

This is the main reason why we need the `player:hud_get_elements()` feature.